pkg/timeutils/utils_test.go

```diff
@@ -8,37 +8,84 @@ import (
-		{"0", "-62167305600"}, // 0 gets parsed year 0
-
+	cases := []struct {
+		in, expected string
+		expectedErr  bool
+	}{
 		// Partial RFC3339 strings get parsed with second precision
-		{"2006-01-02T15:04:05.999999999+07:00", "1136189045"},
-		{"2006-01-02T15:04:05.999999999Z", "1136214245"},
-		{"2006-01-02T15:04:05.999999999", "1136214245"},
-		{"2006-01-02T15:04:05", "1136214245"},
-		{"2006-01-02T15:04", "1136214240"},
-		{"2006-01-02T15", "1136214000"},
-		{"2006-01-02T", "1136160000"},
-		{"2006-01-02", "1136160000"},
-		{"2006", "1136073600"},
-		{"2015-05-13T20:39:09Z", "1431549549"},
+		{"2006-01-02T15:04:05.999999999+07:00", "1136189045.999999999", false},
+		{"2006-01-02T15:04:05.999999999Z", "1136214245.999999999", false},
+		{"2006-01-02T15:04:05.999999999", "1136214245.999999999", false},
+		{"2006-01-02T15:04:05Z", "1136214245.000000000", false},
+		{"2006-01-02T15:04:05", "1136214245.000000000", false},
+		{"2006-01-02T15:04:0Z", "", true},
+		{"2006-01-02T15:04:0", "", true},
+		{"2006-01-02T15:04Z", "1136214240.000000000", false},
+		{"2006-01-02T15:04+00:00", "1136214240.000000000", false},
+		{"2006-01-02T15:04-00:00", "1136214240.000000000", false},
+		{"2006-01-02T15:04", "1136214240.000000000", false},
+		{"2006-01-02T15:0Z", "", true},
+		{"2006-01-02T15:0", "", true},
+		{"2006-01-02T15Z", "1136214000.000000000", false},
+		{"2006-01-02T15+00:00", "1136214000.000000000", false},
+		{"2006-01-02T15-00:00", "1136214000.000000000", false},
+		{"2006-01-02T15", "1136214000.000000000", false},
+		{"2006-01-02T1Z", "1136163600.000000000", false},
+		{"2006-01-02T1", "1136163600.000000000", false},
+		{"2006-01-02TZ", "", true},
+		{"2006-01-02T", "", true},
+		{"2006-01-02+00:00", "1136160000.000000000", false},
+		{"2006-01-02-00:00", "1136160000.000000000", false},
+		{"2006-01-02-00:01", "1136160060.000000000", false},
+		{"2006-01-02Z", "1136160000.000000000", false},
+		{"2006-01-02", "1136160000.000000000", false},
+		{"2015-05-13T20:39:09Z", "1431549549.000000000", false},
 
 		// unix timestamps returned as is
-		{"1136073600", "1136073600"},
-
+		{"1136073600", "1136073600", false},
+		{"1136073600.000000001", "1136073600.000000001", false},
 		// Durations
-		{"1m", fmt.Sprintf("%d", now.Add(-1*time.Minute).Unix())},
-		{"1.5h", fmt.Sprintf("%d", now.Add(-90*time.Minute).Unix())},
-		{"1h30m", fmt.Sprintf("%d", now.Add(-90*time.Minute).Unix())},
+		{"1m", fmt.Sprintf("%d", now.Add(-1*time.Minute).Unix()), false},
+		{"1.5h", fmt.Sprintf("%d", now.Add(-90*time.Minute).Unix()), false},
+		{"1h30m", fmt.Sprintf("%d", now.Add(-90*time.Minute).Unix()), false},
 
 		// String fallback
-		{"invalid", "invalid"},
+		{"invalid", "invalid", false},
+	}
+
+	for _, c := range cases {
+		o, err := GetTimestamp(c.in, now)
+		if o != c.expected ||
+			(err == nil && c.expectedErr) ||
+			(err != nil && !c.expectedErr) {
+			t.Fatalf("wrong value for '%s'. expected:'%s' got:'%s' with error: `%s`", c.in, c.expected, o, err)
```

Comment:
In the interest of running the whole table to see if there are any other errors, maybe
```
t.Fail()
t.Errorf(...)
```

Also, maybe print the value of `expectedErr`, to know if that is what failed the check.